host/crypto/bcrypt/sha.c

```diff
@@ -81,3 +119,64 @@ oe_result_t oe_sha256_final(oe_sha256_context_t* context, OE_SHA256* sha256)
 done:
     return result;
 }
+
+#ifdef OE_WITH_EXPERIMENTAL_EEID
+oe_result_t oe_sha256_save(
+    const oe_sha256_context_t* context,
+    uint32_t* internal_hash,
+    uint32_t* num_hashed)
+{
+    oe_result_t result = OE_INVALID_PARAMETER;
+    oe_sha256_context_impl_t* impl = (oe_sha256_context_impl_t*)context;
+
+    if (!context || !internal_hash || !num_hashed)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    /* This is very brittle; these offsets into the state object may change
```

Comment:
Yes, it could get nasty if the memory layout changes, but I would hope that doesn't happen frequently.

I have no idea about the certification of any of the libraries. I think there's an old version of OpenSSL that does have certification, but I don't know if that covers everything we need. Conversely, it seems to me that BCrypt doesn't actually run in "FIPS mode" by default.